### PR TITLE
Snhut 2258 Staff Profile UI implementaion with mock endpoints

### PR DESCRIPTION
**Pull request by Mubashir-sadaindia**
### Description

- To create a new module/page for displaying and updating staff profiles (Personal and Contact details) with the help of mock endpoints (Which has to be updated when the endpoints are provided by the Backend team)

### Video 
[Sample Demo.webm](https://user-images.githubusercontent.com/118154040/209935774-540937a6-da59-4cd2-9b9e-d8b413e17cee.webm)

#### Screen shots
Screen Shot 2022-12-29 at 3 35 31 PM
Screen Shot 2022-12-29 at 3 35 40 PM

### Other Details

- Sample id `uuid:1234567890`, if needed to navigate and test it out 

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2258)
- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2259)

#### Checklist

- [ ] **Assumptions of User Story met (including acceptance criteria)**
- [ ] **All new necessary Unit tests were CREATED** to promote code coverage
- [ ] **All required unit tests are PASSING**
- [ ] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [ ] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [ ] Ensure configuration and/or build changes documented

### CODE REVIEW

File: apps/toboggan-api/src/staff-profile/staff-profile.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { IStaffProfile } from '@toboggan-ws/toboggan-common';
+
+@Injectable()
+export class StaffProfileService {
+	staffProfile: IStaffProfile;
+
+	constructor() {
+		for (let i = 0; i < 20; i++) {
+			this.staffProfile = {
```
Comment (sada-ajith) [inline apps/toboggan-api/src/staff-profile/staff-profile.service.ts:10]: move this to a mock file which can reused in unit testing context as well

File: apps/toboggan-app/src/app/staff-profile/services/staff-profile.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    });
+    service = TestBed.inject(StaffProfileService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/staff-profile/services/staff-profile.service.spec.ts:22]: add unit tests

File: apps/toboggan-api/src/staff-profile/staff-profile.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      ],
+      controllers: [StaffProfileController],
+      providers: [StaffProfileService]
+    }).compile();
+
+    controller = module.get<StaffProfileController>(StaffProfileController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
```
Comment (sada-ajith) [inline apps/toboggan-api/src/staff-profile/staff-profile.controller.spec.ts:28]: Please add unit tests

**Review by sada-ajith — CHANGES_REQUESTED** [review]
Could you please work in requested changes

File: apps/toboggan-api/src/staff-profile/staff-profile.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      providers: [StaffProfileService],
+    }).compile();
+  });
+
+  beforeEach(async () => {
+    http = module.get<HttpService>(HttpService);
+    service = module.get<StaffProfileService>(StaffProfileService);
+  });
+
+  it('should be defined', () => {
```
Comment (sada-ajith) [inline apps/toboggan-api/src/staff-profile/staff-profile.service.spec.ts:29]: add unit tests for service as well

File: apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
+@use "../../../shared/styles/bootstrap_overrides";
+@import '../../../shared/styles/colors';
+@import "~bootstrap/scss/mixins/breakpoints";
+
+.details-container {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    // padding: 32px;
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.scss:9]: could you please remove this comment?

File: apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  templateUrl: './contact-tab.component.html',
+  styleUrls: ['./contact-tab.component.scss']
+})
+export class ContactTabComponent implements OnChanges {
+  @Input() profile = {} as IStaffProfile;
+
+  editMode = false;
+
+  toolTips = {
+    phoneNumber: 'If you don’t want learners to have your personal phone number, you can generate a new number using a free online masking service and enter that here. Incoming calls or texts will still be routed to your personal phone number.',
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.ts:21]: Please add necessary unit tests for this component

File: apps/toboggan-app/src/app/staff-profile/components/personal-tab/personal-tab.component.ts [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+      bio: bio
+    } = profile;
+
+    this.personalDetailsForm.patchValue({
+      firstName,
+      lastName,
+      preferredName,
+      preferredPronoun,
+      bio
+    });
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/staff-profile/components/personal-tab/personal-tab.component.ts:70]: Please add necessary unit tests for this component

File: apps/toboggan-app/src/app/staff-profile/pages/staff-profile-main-page/staff-profile-main-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+  selector: 'toboggan-ws-staff-profile-main-page',
+  templateUrl: './staff-profile-main-page.component.html',
+  styleUrls: ['./staff-profile-main-page.component.scss']
+})
+export class StaffProfileMainPageComponent implements OnInit {
+  id: any;
+  profile = {} as IStaffProfile;
+  constructor(private route: ActivatedRoute, private staffService: StaffProfileService) { }
+
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/staff-profile/pages/staff-profile-main-page/staff-profile-main-page.component.ts:16]: Please add necessary unit tests for this component

**Review by sada-ajith — CHANGES_REQUESTED** [review]
Please check my notes

File: apps/toboggan-api/src/staff-profile/staff-profile.service.ts (hunk 1 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-api/src/staff-profile/staff-profile.service.ts]: done

File: apps/toboggan-app/src/app/staff-profile/services/staff-profile.service.spec.ts (hunk 2 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-app/src/app/staff-profile/services/staff-profile.service.spec.ts:22]: done

File: apps/toboggan-api/src/staff-profile/staff-profile.service.spec.ts (hunk 4 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-api/src/staff-profile/staff-profile.service.spec.ts:29]: done

File: apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.scss (hunk 5 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.scss]: done

File: apps/toboggan-app/src/app/staff-profile/components/personal-tab/personal-tab.component.ts (hunk 7 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-app/src/app/staff-profile/components/personal-tab/personal-tab.component.ts:70]: done

File: apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.ts (hunk 6 above)
Comment (Mubashir-sadaindia) [reply apps/toboggan-app/src/app/staff-profile/components/contact-tab/contact-tab.component.ts:21]: done

**Review by sada-ajith — APPROVED** [review]

**Pull request merged**